File: testsuite/tests/constructorcall.cpp
class fooBar;

class foo{
public:
  foo(int i){
    val = i;
  }
  int val;
};

int bar(foo f = foo(0)){
  return f.val;
}


int bar(fooBar f = fooBar(0));

class ConsInitList : public foo{
public:
  ConsInitList(int i) : foo(i) {}
};

File: testsuite/results/constructorcall.nim
discard "forward decl of fooBar"
type
  foo* {.bycopy.} = object
    val*: cint


proc constructfoo*(i: cint): foo {.constructor.}
proc bar*(f: foo = constructfoo(0)): cint =
  discard

proc bar*(f: fooBar = constructfooBar(0)): cint
type
  ConsInitList* {.bycopy.} = object of foo


proc constructConsInitList*(i: cint): ConsInitList {.constructor.}